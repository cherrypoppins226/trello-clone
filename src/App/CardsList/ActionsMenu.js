import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Divider from "@material-ui/core/Divider";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";
import grey from "@material-ui/core/colors/grey";
import { connect } from "react-redux";

import * as actions from "../actions";
import * as labels from "../labels";

const styles = {
  root: {
    width: 300,
    "& [role='heading']": {
      paddingTop: 8,
      color: grey[700]
    },
    "& [role='menu']": {
      "& *": {
        fontWeight: 700
      },
      "& hr": {
        margin: "5px 10px"
      },
      "& [role='menuitem']": {
        padding: 3,
        paddingLeft: 10,
        "&:hover": {
          backgroundColor: "rgb(41, 143, 202)",
          "& p": { color: "white" }
        }
      }
    }
  }
};

const ActionsMenu = props => {
  return (
    <Paper
      className={props.classes.root}
      aria-describedby={labels.cardsListActionsMenuDescription.id}
    >
      <Typography role="heading" align="center">
        List Actions
      </Typography>
      <MenuList dense={true} onClick={() => props.finishEditList()}>
        {[
          "",
          "Add Card...",
          "Copy List...",
          "Move List...",
          "Watch",
          "",
          "Sort By",
          "",
          "Move All Cards in This List...",
          "Archive All Cards in This List...",
          "",
          "Archive This List"
        ].map((item, idx) => {
          if (!Boolean(item)) return <Divider key={idx} />;
          return (
            <MenuItem key={idx}>
              <Typography> {item} </Typography>
            </MenuItem>
          );
        })}
      </MenuList>
    </Paper>
  );
};

const mapDispatchToProps = dispatch => ({
  finishEditList: () => dispatch(actions.finishEditList())
});

export default connect(null, mapDispatchToProps)(
  withStyles(styles)(ActionsMenu)
);
