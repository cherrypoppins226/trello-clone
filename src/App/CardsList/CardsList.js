import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { fileAbsolute } from "paths.macro";

import Header from "./Header";
import Cards from "./Cards";
import { button } from "../styles";
import { moduleName } from "../utils";
import { mapDispatchToProps } from "../redux";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    background: "rgb(226, 228, 230)",
    "& > button": {
      ...button,
      justifyContent: "left",
      fontWeight: 400
    }
  },
  header: {
    margin: 5
  }
};

const View = props => {
  return (
    <Paper
      data-listid={props.list.id}
      component="section"
      elevation={1}
      className={props.classes.root}
    >
      <Header
        className={props.classes.header}
        listId={props.list.id}
        listTitle={props.list.title}
      />
      <div style={{ overflowY: "scroll" }}>
        <Cards cards={props.list.cards} />
      </div>
      <Button onClick={() => props.actions.addCard(props.list.id)}>
        Add a card...
      </Button>
    </Paper>
  );
};

const mapStateToProps = (state, ownProps) => ({
  list: state.lists.find(list => list.id === ownProps.list.id)
});

const Container = connect(mapStateToProps, mapDispatchToProps)(
  withStyles(styles)(View)
);

Container.displayName = moduleName(fileAbsolute);

export const listType = PropTypes.shape({
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired
});

Container.propTypes = {
  list: listType
};

export default Container;
