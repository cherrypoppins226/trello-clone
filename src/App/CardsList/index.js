export { default, listType } from "./CardsList";
