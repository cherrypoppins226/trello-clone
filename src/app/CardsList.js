import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";
import PropTypes from "prop-types";
import { fileAbsolute } from "paths.macro";
import { graphql } from "react-apollo";
import { compose, setDisplayName, setPropTypes } from "recompose";

import AppState from "../App.state";
import Header from "./cardsList/Header";
import Cards from "./cardsList/Cards";
import { queries } from "../cosmos/apollo/schema";
import { button } from "./styles";
import { makeFixtures, handleGraphQLResponse, moduleName } from "../utils";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    background: "rgb(226, 228, 230)",
    "& > button": {
      ...button,
      justifyContent: "left",
      fontWeight: 400
    }
  },
  header: {
    margin: 5
  }
};

const CardsList = ({ classes, id, addCard, data: { list, variables } }) => {
  return (
    <Paper
      data-listid={id}
      component="section"
      elevation={1}
      className={classes.root}
    >
      <Header
        className={classes.header}
        listId={list.id}
        listTitle={list.title}
      />
      <div style={{ overflowY: "scroll" }}>
        <Cards cards={list.cards} />
      </div>
      <Button
        onClick={() => {
          const listQuery = { query: queries.list, variables };
          addCard({
            update: (proxy, { data: { addCard } }) => {
              const data = proxy.readQuery(listQuery);
              data.list.cards.push(addCard);
              proxy.writeQuery({ ...listQuery, data });
            }
          });
        }}
      >
        Add a card...
      </Button>
    </Paper>
  );
};

const Component = compose(
  setDisplayName(moduleName(fileAbsolute)),
  setPropTypes({
    id: PropTypes.number.isRequired
  }),
  graphql(queries.list, {
    options: props => ({ variables: { id: props.id } })
  }),
  handleGraphQLResponse(),
  graphql(queries.addCard, {
    name: "addCard",
    options: props => ({
      variables: { listId: props.id, title: "Title..." },
      optimisticResponse: {
        __typename: "Mutation",
        addCard: {
          __typename: "Card",
          id: -1,
          title: "Title..."
        }
      }
    })
  }),
  withStyles(styles)
)(CardsList);

export const fixtures = makeFixtures(Component, {
  default: {
    props: {
      id: 1
    },
    stores: {
      appState: new AppState()
    }
  }
});

export default Component;
