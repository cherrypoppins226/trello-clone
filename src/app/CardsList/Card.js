import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";
import grey from "@material-ui/core/colors/grey";
import Create from "@material-ui/icons/Create";
import merge from "deepmerge";
import { fileAbsolute } from "paths.macro";
import { inject } from "mobx-react";

import AppState from "../../App.state";
import { makeFixtures, renderLabels, labelId, moduleName } from "../../utils";
import { buttonIconSmall } from "../styles";

const modulePath = moduleName(fileAbsolute);

export const labels = {
  editCard: {
    id: labelId(modulePath, "edit-card"),
    text: "Edit card"
  },
  quickEditCard: {
    id: labelId(modulePath, "quick-edit-card"),
    text: "Quickly edit card"
  }
};

const styles = {
  root: {
    padding: 4,
    paddingLeft: 8,
    display: "flex",
    alignItems: "flex-start",
    justifyContent: "space-between",
    cursor: "pointer",
    "&:hover": {
      background: grey[200],
      "& button": {
        visibility: "visible"
      }
    },
    "& button": merge(buttonIconSmall, {
      visibility: "hidden",
      paddingBottom: 0,
      "&:hover": {
        visibility: "visible"
      }
    })
  }
};

const Card = ({ classes, appState, card }) => {
  // TODO: Use <Card /> from Material UI
  return (
    <Paper
      data-cardid={card.id}
      elevation={1}
      onClick={e => appState.startCardEdit(card)}
      className={classes.root}
      aria-labelledby={labels.editCard.id}
    >
      {renderLabels(labels)}
      <Typography data-testid="card-title">{card.title}</Typography>
      <button
        aria-labelledby={labels.quickEditCard.id}
        onClick={e => {
          e.stopPropagation();
          const box = e.currentTarget.parentElement.getBoundingClientRect();
          const { top, left, bottom, right } = box;
          appState.startQuickCardEdit({
            ...card,
            anchorElementBox: { top, left, bottom, right }
          });
        }}
      >
        <Create />
      </button>
    </Paper>
  );
};

const Component = withStyles(styles)(inject("appState")(Card));

Component.displayName = modulePath;

export const types = {
  card: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
  })
};

Component.propTypes = {
  card: types.card.isRequired
};

export const fixtures = makeFixtures(Component, {
  default: {
    props: {
      card: { id: 1, title: "Ut sunt qui amet." }
    },
    stores: {
      appState: new AppState()
    }
  }
});

export default Component;
