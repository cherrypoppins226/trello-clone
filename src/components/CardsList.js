import React from "react";
import { findDOMNode } from "react-dom";
import path from "path";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";
import Typography from "material-ui/Typography";
import Paper from "material-ui/Paper";
import grey from "material-ui/colors/grey";
import PropTypes from "prop-types";
import MoreHoriz from "@material-ui/icons/MoreHoriz";
import TextArea from "./TextArea";
import CardsListCard from "./CardsListCard";
import * as Labels from "./labels";

const styles = {
  container: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "100%",
    background: "rgb(226, 228, 230)",
    "& > button": {
      justifyContent: "left",
      textTransform: "none"
    },
    "& ul": {
      padding: 0,
      margin: 0,
      height: "100%"
    }
  },
  listHeader: {
    display: "flex",
    flexShrink: 0,
    justifyContent: "space-between",
    margin: 5,
    "& [role='heading']": {
      flexBasis: "100%",
      fontWeight: 700,
      paddingBottom: 0,
      paddingTop: 3,
      paddingLeft: 5,
      marginRight: 8
    },
    "& p[role='heading']": {
      cursor: "pointer"
    },
    "& button": {
      alignSelf: "flex-start",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      borderRadius: 4,
      padding: 3,
      "&:hover": {
        background: "#D6D6D6"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

const onOutsideClick = (node, handler) => {
  const onClick = event => {
    if (!node.contains(event.target)) {
      document.body.removeEventListener("click", onClick);
      handler(event);
    }
  };
  document.body.addEventListener("click", onClick);
};

const Title = class extends React.Component {
  constructor(props) {
    super(props);
    this.state = { editing: false };
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.editing)
      onOutsideClick(findDOMNode(this), _ => this.setState({ editing: false }));
  }

  render() {
    const props = this.state.editing
      ? {
          component: TextArea,
          value: this.props.text
        }
      : {
          onClick: _ => this.setState({ editing: true }),
          children: this.props.text
        };
    return <Typography role="heading" {...props} />;
  }
};

const View = class extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.cards.length,
      cards: props.cards.map((description, idx) =>
        this.newCard(idx, description)
      )
    };
  }

  newCard(id, description = undefined) {
    return (
      <CardsListCard
        key={id}
        description={description}
        onEditCard={this.props.onEditCard}
      />
    );
  }

  addCard(e) {
    e.preventDefault();
    this.setState((prevState, props) => ({
      counter: prevState.counter + 1,
      cards: [...prevState.cards, this.newCard(prevState.counter + 1)]
    }));
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.cards.length < this.state.cards.length)
      this.cardsListEnd.scrollIntoView();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper
        component="section"
        elevation={1}
        className={classes.container}
        data-testid="CardsList"
      >
        <div className={classes.listHeader}>
          <Title text={this.props.title} />
          <button
            onClick={e => this.props.onEditList(e.currentTarget)}
            aria-describedby={Labels.cardsListActionsMenu.id}
            aria-haspopup={true}
          >
            <MoreHoriz />
          </button>
        </div>
        <div style={{ overflowY: "scroll" }}>
          <ul>{this.state.cards}</ul>
          {/* The sole purpose of this div is to allow scrolling to bottom */}
          <div
            style={{ float: "left", clear: "both" }}
            ref={node => (this.cardsListEnd = node)}
          />
        </div>
        <Button onClick={this.addCard.bind(this)}>Add a card...</Button>
      </Paper>
    );
  }
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired,
  onEditList: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
