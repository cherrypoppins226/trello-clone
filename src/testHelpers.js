import { render as renderer } from "react-testing-library";
import { createContext } from "react-cosmos-loader";

export const render = (fixture, proxies = []) => {
  const { mount, ...context } = createContext({ renderer, fixture, proxies });

  return (async () => {
    await mount();
    return { ...context.getWrapper(), ...context, rerender: mount };
  })();
};

export const NativeEvents = {
  mouse: {
    click: new MouseEvent("click", { bubbles: true, cancelable: true })
  }
};

export const getByRole = (node, role) => {
  return node.querySelector(`[role="${role}"]`);
};

export const getByAriaLabelled = (node, id) => {
  return node.querySelector(`[aria-labelledby="${id}"]`);
};

export const getByAriaDescribed = (node, id) => {
  return node.querySelector(`[aria-describedby="${id}"]`);
};
