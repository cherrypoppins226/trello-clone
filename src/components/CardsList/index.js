import React from "react";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";
import Paper from "material-ui/Paper";
import grey from "material-ui/colors/grey";
import PropTypes from "prop-types";
import MoreHoriz from "@material-ui/icons/MoreHoriz";
import Title from "./Title";
import Card from "./Card";
import * as Labels from "../labels";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "100%",
    background: "rgb(226, 228, 230)",
    "& > button": {
      justifyContent: "left",
      textTransform: "none"
    },
    "& ul": {
      padding: 0,
      margin: 0,
      height: "100%",
      "& li": {
        margin: 8,
        marginTop: 2,
        cursor: "pointer"
      }
    }
  },
  listHeader: {
    display: "flex",
    flexShrink: 0,
    justifyContent: "space-between",
    margin: 5,
    "& [role='heading']": {
      flexBasis: "100%",
      fontWeight: 700,
      paddingBottom: 0,
      paddingTop: 3,
      paddingLeft: 5,
      marginRight: 8
    },
    "& p[role='heading']": {
      cursor: "pointer"
    },
    "& button": {
      alignSelf: "flex-start",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      borderRadius: 4,
      padding: 3,
      "&:hover": {
        background: "#D6D6D6"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

class CardsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.cards.length,
      cards: props.cards.map((card, idx) => ({
        id: idx,
        description: card
      }))
    };
  }

  addCard(e) {
    e.preventDefault();
    this.setState((prevState, props) => ({
      counter: prevState.counter + 1,
      cards: [
        ...prevState.cards,
        { id: prevState.counter + 1, description: undefined }
      ]
    }));
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.cards.length < this.state.cards.length)
      this.cardsListEnd.scrollIntoView();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper
        component="section"
        elevation={1}
        className={classes.root}
        data-testid="CardsList"
      >
        <div className={classes.listHeader}>
          <Title role="heading" text={this.props.title} />
          <button
            onClick={e => this.props.onEditList(e.currentTarget)}
            aria-labelledby={Labels.cardsListActionsMenu.id}
            aria-haspopup={true}
          >
            <MoreHoriz />
          </button>
        </div>
        <div style={{ overflowY: "scroll" }}>
          <ul>
            {this.state.cards.map(({ id, description }) => (
              <li key={id}>
                <Card
                  description={description}
                  onQuickEditCard={this.props.onQuickEditCard}
                  onEditCard={this.props.onEditCard}
                />
              </li>
            ))}
          </ul>
          {/* The sole purpose of this div is to allow scrolling to bottom */}
          <div
            style={{ float: "left", clear: "both" }}
            ref={node => (this.cardsListEnd = node)}
          />
        </div>
        <Button onClick={this.addCard.bind(this)}>Add a card...</Button>
      </Paper>
    );
  }
}

const View = withStyles(styles)(CardsList);

View.propTypes = {
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired,
  onEditList: PropTypes.func.isRequired,
  onQuickEditCard: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired
};

export default View;
