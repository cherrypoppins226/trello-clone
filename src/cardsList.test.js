import React from "react";
import { Simulate } from "react-testing-library";
import { render } from "./testHelpers.js";
import CardsList, { LIST_ACTIONS_MENU_LABEL } from "./cardsList";

jest.mock("./cardsListCard", () => () => "CardsListCard");
jest.mock("./cardsListActionsMenu", () => () => "CardsListActionMenu");

const title = "List title";
const cards = ["card1", "card2"];

it("initial render", () => {
  const { container } = render(
    <CardsList title={title} cards={cards} onEditCard={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});

describe("", () => {
  beforeAll(() => {
    jest.unmock("./cardsListCard");
    jest.resetModules();
  });

  afterAll(() => {
    jest.mock("./cardsListCard");
    jest.resetModules();
  });

  it("adds a card", () => {
    const CardsList = require("./cardsList").default;
    const { getByText, getByTestId } = render(
      <CardsList
        title="Title"
        cards={["card1", "card2"]}
        onEditCard={jest.fn()}
      />
    );
    const liveList = getByTestId("cards-list");
    const countBefore = liveList.childElementCount;
    const lastBefore = liveList.lastElementChild;
    Simulate.click(getByText("Add a card..."));
    expect(liveList.childElementCount).toBe(countBefore + 1);
    expect(liveList.lastElementChild).not.toBe(lastBefore);
    expect(liveList.lastElementChild.tagName).toBe(lastBefore.tagName);
  });
});

describe("", () => {
  beforeAll(() => {
    jest.unmock("./cardsListActionsMenu");
    jest.resetModules();
  });

  afterAll(() => {
    jest.mock("./cardsListActionsMenu");
    jest.resetModules();
  });

  it("opens actions menu", () => {
    const CardsList = require("./cardsList").default;
    const { container } = render(
      <CardsList
        title="Title"
        cards={["card1", "card2"]}
        onEditCard={jest.fn()}
      />
    );
    const actionsMenuButton = container.querySelector(
      `[aria-labelledby="${LIST_ACTIONS_MENU_LABEL}"]`
    );
    Simulate.click(actionsMenuButton);
    expect(document.body).toMatchSnapshot();
  });
});
