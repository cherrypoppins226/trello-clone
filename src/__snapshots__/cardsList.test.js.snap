// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
<div>
  <section
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded View-container-1"
  >
    <div
      class="View-listHeader-2"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        role="heading"
      >
        List title
      </p>
      <button
        aria-haspopup="true"
        aria-labelledby="open-list-actions-menu"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </g>
        </svg>
      </button>
      CardsListActionMenu
    </div>
    <div
      style="overflow-y: scroll;"
    >
      <ul
        data-testid="cards-list"
      >
        CardsListCard
        CardsListCard
      </ul>
      <div
        style="float: left; clear: both;"
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Add a card...
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </section>
</div>
`;

exports[`opens actions menu 1`] = `
<body
  style="overflow: hidden; padding-right: 0px;"
>
  <div
    class="MuiModal-root View-container-230"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiPopover-paper"
      role="document"
      style="opacity: 1; top: 16px; left: 16px;"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
        role="heading"
      >
        List Actions
      </p>
      <ul
        class="MuiList-root MuiList-dense MuiList-padding"
        role="menu"
      >
        <hr
          class="MuiDivider-root"
          tabindex="0"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Add Card...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Copy List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Move List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Watch
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
          tabindex="-1"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Sort By
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
          tabindex="-1"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Move All Cards in This List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Archive All Cards in This List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
          tabindex="-1"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Archive This List
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</body>
`;
